tct/src/internal/hash.rs

```diff
@@ -103,14 +115,98 @@ impl Hash {
         Self(Fp256::new(BigInteger256(bytes)))
     }
 
+    /// Hash an individual item to be inserted into the tree.
+    #[inline]
+    pub fn of(item: Commitment) -> Hash {
+        Self(hash_1(&DOMAIN_SEPARATOR, item.into()))
+    }
+
     /// Construct a hash for an internal node of the tree, given its height and the hashes of its
     /// four children.
     #[inline]
     pub fn node(height: u8, Hash(a): Hash, Hash(b): Hash, Hash(c): Hash, Hash(d): Hash) -> Hash {
         let height = Fq::from_le_bytes_mod_order(&height.to_le_bytes());
-        Hash(poseidon377::hash_4(
-            &(*DOMAIN_SEPARATOR + height),
-            (a, b, c, d),
-        ))
+        Hash(hash_4(&(*DOMAIN_SEPARATOR + height), (a, b, c, d)))
+    }
+}
+
+#[cfg(feature = "sqlx")]
+mod sqlx_impls {
+    use decaf377::{FieldExt, Fq};
+    use sqlx::{Database, Decode, Encode, Postgres, Type};
+    use thiserror::Error;
+
+    use super::*;
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Error)]
+    #[error("expected exactly 32 bytes")]
+    struct IncorrectLength;
+
+    impl<'r> Decode<'r, Postgres> for Hash {
+        fn decode(
+            value: <Postgres as sqlx::database::HasValueRef<'r>>::ValueRef,
+        ) -> Result<Self, sqlx::error::BoxDynError> {
+            let bytes: [u8; 32] = <Vec<u8> as sqlx::Decode<Postgres>>::decode(value)?
+                .try_into()
+                .map_err(|_| IncorrectLength)?;
+            Ok(Hash(Fq::from_bytes(bytes)?))
+        }
+    }
+
+    impl<'q> Encode<'q, Postgres> for Hash {
+        fn encode_by_ref(
+            &self,
+            buf: &mut <Postgres as sqlx::database::HasArguments<'q>>::ArgumentBuffer,
+        ) -> sqlx::encode::IsNull {
+            let bytes = self.0.to_bytes();
+            <&[u8] as sqlx::Encode<Postgres>>::encode(&bytes[..], buf)
+        }
+    }
+
+    impl Type<Postgres> for Hash {
+        fn type_info() -> <Postgres as Database>::TypeInfo {
+            <[u8] as Type<Postgres>>::type_info()
+        }
+    }
+}
```

Comment:
This code should also not be added, since we don't use Postgres any more.

Reply:
Removed in tct-integration-tct-only.